infrastructure/code/code_test.go

```diff
@@ -186,11 +203,26 @@ func Test_Scan(t *testing.T) {
 
 	t.Run("Shouldn't run if Sast is disabled", func(t *testing.T) {
 		c := testutil.UnitTest(t)
+		ctrl := gomock.NewController(t)
+		mockEngine := mocks.NewMockEngine(ctrl)
+		mockConfig := mocks.NewMockConfiguration(ctrl)
+		c.SetEngine(mockEngine)
 
-		scanner := New(c, performance.NewInstrumentor(), &snyk_api.FakeApiClient{CodeEnabled: false}, newTestCodeErrorReporter(), nil, notification.NewNotifier(), &FakeCodeScannerClient{})
 		tempDir, _, _ := setupIgnoreWorkspace(t)
 
-		c.Engine().GetConfiguration().Set(code_workflow.ConfigurationSastSettings, &sast_contract.SastResponse{SastEnabled: false})
+		mockEngine.EXPECT().GetConfiguration().Return(mockConfig).AnyTimes()
+		mockConfig.EXPECT().GetString(gomock.Any()).Return("").AnyTimes()
+		mockConfig.EXPECT().Set(gomock.Any(), gomock.Any()).AnyTimes()
+
+		// Mock Clone() to return a cloned configuration
+		clonedConfig := mocks.NewMockConfiguration(ctrl)
+		clonedConfig.EXPECT().Set(gomock.Any(), gomock.Any()).AnyTimes()
+		clonedConfig.EXPECT().GetWithError(code_workflow.ConfigurationSastSettings).Return(&sast_contract.SastResponse{SastEnabled: false}, nil).AnyTimes()
+		mockConfig.EXPECT().Clone().Return(clonedConfig).AnyTimes()
+
+		scanner := New(c, performance.NewInstrumentor(), &snyk_api.FakeApiClient{CodeEnabled: false}, newTestCodeErrorReporter(), nil, notification.NewNotifier(), &FakeCodeScannerClient{})
```

Comment:
apart from the mock expectation there is no real check to verify the scanner does not run, right?